app.rb

```diff
@@ -21,20 +21,26 @@ class App < Sinatra::Base
     erb :index
   end
 
-  get '/user_profile/' do
-    erb :user_profile
-  end
-
   get '/sessions/new' do
     erb :sign_in
   end
 
   post '/sessions/new' do
     email = params[:inputEmail]
     password = params[:inputPassword]
-    user = User.create(email, password)
-    session[:user] = user
-    redirect '/'
+    user = User.authenticate(email, password)
+    if user
+      redirect '/user_profile'
+    else
+      puts("Wrong Username And/or Password")
```

Comment:
Could we store this in a session so that it can appear on the redirected page as a message?